### PR TITLE
Assignment04

### PR DESCRIPTION
**Pull request by ayemoekyaw99**

### CODE REVIEW

File: assignment_04/app/Contracts/Dao/MajorDaoInterface.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
```
Comment (ingyinmaycgm) [inline assignment_04/app/Contracts/Dao/MajorDaoInterface.php:1]: Check previous assignment PR review in this assignment and fix if still incorrect.
**For the whole project files**

File: assignment_04/public/js/major.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                                                <td>${response.major.id}</td>
+                                                <td>${response.major.name}</td>
+                                                <td>
+                                                    <button class="deleteMajor btn btn-danger btn-sm" data-id="${response.major.id}">Delete</button>
+                                                </td>
+                                            </tr>
+                                            `);
+                    $('#createMajorForm')[0].reset();
+                }
+            },
```
Comment (ingyinmaycgm) [inline assignment_04/public/js/major.js:29]: How about error condition? 
Please make validation and print error message when validation fail.

File: assignment_04/resources/views/major/create.blade.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
+@extends('layout.app')
```
Comment (ingyinmaycgm) [inline assignment_04/resources/views/major/create.blade.php:1]: May be you don't need this file in this tutorial.

File: assignment_04/app/Http/Controllers/MajorController.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    /**
+     *show create
+     *
+     *return @string
+     */
+    public function create()
+    {
+        return view('major.create');
+    }
```
Comment (ingyinmaycgm) [inline assignment_04/app/Http/Controllers/MajorController.php:43]: Remove this unless it is used.

File: assignment_04/routes/web.php [hunk 5]
```diff
@@ -12,7 +15,19 @@
...
-});
+Route::get('/', [StudentController::class, 'index'])->name('students#list');
+Route::get('/student/create', [StudentController::class, 'createStudent'])->name('student#create');
+Route::post('/student/store', [StudentController::class, 'store'])->name('student#store');
+Route::get('/student/delete/{id}', [StudentController::class, 'delete'])->name('student#delete');
+Route::get('/student/edit/{id}', [StudentController::class, 'edit'])->name('student#edit');
+Route::post('/student/update/{id}', [StudentController::class, 'updateStudent'])->name('student#update');
+
+Route::get('/student/export', [ExcelController::class, 'export'])->name('student#export');
+Route::post('/student/import', [ExcelController::class, 'import'])->name('student#import');
```
Comment (ingyinmaycgm) [inline assignment_04/routes/web.php:33]: Check all file and add eof.

File: assignment_04/app/Dao/StudentDao.php [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                    ->orWhere('students.email', '=', $key)
+                    ->orWhere('students.phone', '=', $key)
+                    ->orWhere('students.address', 'like', '%' . $key . '%')
+                    ->orWhere('majors.name', 'like', '%' . $key . '%');
+            })->get();
+
+        //   $students = DB::table('students')
+        //        ->leftJoin('majors', 'students.major_id', '=', 'majors.id')
+        //        ->when(request('searchKey'),function($query){
+        //                        $key=request('searchKey');
```
Comment (thinpyantsoe) [inline assignment_04/app/Dao/StudentDao.php:33]: Remove no use code.

File: assignment_04/app/Dao/StudentDao.php [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    /**
+   * Student update
+
+   * @param mixed $data
+   * @return void
+   */
+    public function update(array $data, int $id): void
+    {
+        $student = Student::find($id);
+        $student->update($data);
```
Comment (thinpyantsoe) [inline assignment_04/app/Dao/StudentDao.php:93]: Student::where('id', $id)->update($data);

File: assignment_04/app/Exports/StudentExport.php [hunk 8]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Exports;
+
```
Comment (thinpyantsoe) [inline assignment_04/app/Exports/StudentExport.php:4]: Remove empty line.

File: assignment_04/app/Exports/StudentExport.php [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+class StudentExport implements FromCollection, WithHeadings
+{
+    /**
+     * @return \Illuminate\Support\Collection
+     */
+    public function collection()
+    {
+        return Student::select("id", "name", "major_id", "phone", "email", "address")->get();
+    }
```
Comment (thinpyantsoe) [inline assignment_04/app/Exports/StudentExport.php:18]: Add Line Enter.

**Review by thinpyantsoe — APPROVED** [review]

**Pull request merged**